core/client_test.go

```diff
@@ -47,6 +48,78 @@ func TestClient_StartBlockSubscription_And_GetBlock(t *testing.T) {
 	}
 
 	// unsubscribe to event channel
-	err = client.Unsubscribe(ctx, "NewBlock/Events", types.QueryForEvent(types.EventNewBlock).String())
+	require.NoError(t, client.Unsubscribe(ctx, newBlockSubscriber, types.QueryForEvent(types.EventNewBlock).String()))
+	require.NoError(t, client.Stop())
+}
+
+func TestRemoteClientLifecycle(t *testing.T) {
+	remote := StartMockNode()
+
+	protocol, ip := getRemoteEndpoint(remote)
+
+	client, err := NewRemote(protocol, ip)
+	require.NoError(t, err)
+
+	require.NoError(t, client.Start())
+	require.NoError(t, client.Stop())
+	require.NoError(t, remote.Stop())
+}
+
+func TestRemoteClient_Status(t *testing.T) {
+	remote := StartMockNode()
+	protocol, ip := getRemoteEndpoint(remote)
+	client, err := NewRemote(protocol, ip)
 	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// client must be `started` before performing requests
+	err = client.Start()
+	require.NoError(t, err)
+
+	status, err := client.Status(ctx)
+	require.NoError(t, err)
+	assert.NotNil(t, status)
+
+	require.NoError(t, client.Stop())
+	require.NoError(t, remote.Stop())
+}
+
+func TestRemoteClient_StartBlockSubscription_And_GetBlock(t *testing.T) {
+	remote := StartMockNode()
+	protocol, ip := getRemoteEndpoint(remote)
+	client, err := NewRemote(protocol, ip)
+	require.NoError(t, err)
```

Comment:
Let's move this into `MockRemoteClient` and rename `MockClient` to `MockEmbeddedClient` to keep things well structured and reusable.

Reply:
Pls check latest commit.